plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/UnsafeIntrinsics.java

```diff
@@ -102,25 +103,25 @@ private static void registerCompareAndExchangeIntrinsics(final CompilationContex
         // todo: research to discover if some platforms handle misalignment correctly, make it switchable
 
         intrinsics.registerIntrinsic(unsafeDesc, "compareAndExchangeInt", getCompareAndExchangeDesc(classContext, BaseTypeDescriptor.I),
-            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, VOLATILE, STRONG));
+            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, GlobalAcquire, STRONG));
         intrinsics.registerIntrinsic(unsafeDesc, "compareAndExchangeIntAcquire", getCompareAndExchangeDesc(classContext, BaseTypeDescriptor.I),
-            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, ACQUIRE, STRONG));
+            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, SingleAcquire, STRONG));
         intrinsics.registerIntrinsic(unsafeDesc, "compareAndExchangeIntRelease", getCompareAndExchangeDesc(classContext, BaseTypeDescriptor.I),
-            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, RELEASE, STRONG));
+            (builder, instance, target, arguments) -> doCompareAndExchange(ctxt, builder, arguments, SingleOpaque, STRONG));
```

Comment:
According to `VarHandle`, this is `SinglePlain` on read and `GlobalRelease` on write.